### PR TITLE
feat(csv-parser): created a csv parser to format the csv into an expected schema

### PR DESCRIPTION
**Pull request by Asalem1**
This PR aims to add a custom CSV parser to ensure that raw CSV data predictably returns Schema data in order to make bucket metadata onto the Flows query builder.
live-data

### CODE REVIEW

File: ui/src/shared/actions/schemaThunks.ts [hunk 1]
```diff
@@ -3,14 +3,15 @@ import {Dispatch} from 'react'
 // import {fromFlux as parse} from '@influxdata/giraffe'
 
 // API
-// import {runQuery} from 'src/shared/apis/query'
+import {runQuery} from 'src/shared/apis/query'
```
Comment (drdelambre) [inline ui/src/shared/actions/schemaThunks.ts:6]: this isn't being used

File: ui/src/shared/parsing/flux/schemaParser.test.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      },
+    })
+  })
+
+  it('throws when no metadata is present', () => {
+    expect(() => {
+      parseResponse(RESPONSE_NO_METADATA)
+    }).toThrow()
+  })
+})
```
Comment (drdelambre) [inline ui/src/shared/parsing/flux/schemaParser.test.ts:110]: mmmmm the feeeeeeeels

File: ui/src/shared/actions/schemaThunks.ts (hunk 1 above)
Comment (Asalem1) [reply ui/src/shared/actions/schemaThunks.ts:6]: as in we shouldn't be using this to run the query? It's used here:

https://github.com/influxdata/influxdb/pull/19450/files/0e33e60063b1fab75c9186be801d5a26cf358532#diff-7003025bc3963a7870d490547eb9c4c0R47

Comment (drdelambre) [reply ui/src/shared/actions/schemaThunks.ts:6]: nm then i got confused

**Pull request merged**